garak/generators/huggingface.py

```diff
@@ -553,21 +608,23 @@ def _call_model(
             return [re.sub("^" + re.escape(prompt), "", i) for i in text_output]
 
 
-class LLaVA(Generator):
+class LLaVA(Generator, HFCompatible):
     """Get LLaVA ([ text + image ] -> text) generations"""
 
     DEFAULT_PARAMS = Generator.DEFAULT_PARAMS | {
+        "max_tokens": 4000,
         # "exist_tokens + max_new_tokens < 4K is the golden rule."
         # https://github.com/haotian-liu/LLaVA/issues/1095#:~:text=Conceptually%2C%20as%20long%20as%20the%20total%20tokens%20are%20within%204K%2C%20it%20would%20be%20fine%2C%20so%20exist_tokens%20%2B%20max_new_tokens%20%3C%204K%20is%20the%20golden%20rule.
-        "max_tokens": 4000,
-        # consider shifting below to kwargs or llava_kwargs that is a dict to allow more customization
-        "torch_dtype": torch.float16,
-        "low_cpu_mem_usage": True,
-        "device_map": "cuda:0",
+        "hf_args": {
+            "torch_dtype": "float16",
+            "low_cpu_mem_usage": True,
```

Comment:
is this already covered by `HFCompatible`? prefer to leave this default in one place

Reply:
`HFCompatible` does not provide defaults and the `_gather_hf_params()` reference to `low_cpu_mem_usage` simple addition to a list to ensure this options is passed if the `Callable` signatures do not provide for it explicitly.

Reply:
See earlier comment about multiple places where we defaults for `torch_dtype`

Reply:
Agree -- it'd be ideal to set a default for all HF classes (I think float16 is a reasonable default, but I could see arguments for "auto")